Devservices fail when docker is not available/supported

Sometimes docker is not available or it is not possible to start containers (e.g. CI pipelines in gitlab, localstack might be provided by as a service) 

The documentation https://docs.quarkiverse.io/quarkus-amazon-services/dev/dev-services.html states:

> Dev Services for Amazon Services is automatically enabled for each extensions added to pom.xml except in the following situations:
> 
> - quarkus.devservices.enabled is set to false
> - devservices.enabled is set to false per extension (eg. quarkus.s3.devservices.enabled=false)
> - the endpoint-override is configured (eg. quarkus.s3.endpoint-override=http://localhost:4566)
> - **Your environment does not support Docker**

Reproduce:
Create a project with a quarkus-amazon-services dependency, shutdown docker and run `quarkus dev`.

```
2023-07-17 21:57:02,654 INFO  [io.qua.ama.lam.run.MockEventServer] (build-15) Mock Lambda Event Server Started
2023-07-17 21:57:03,157 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-13) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2023-07-17 21:57:03,424 WARN  [org.tes.doc.DockerClientProviderStrategy] (build-13) DOCKER_HOST unix:///var/run/docker.sock is not listening
2023-07-17 21:57:03,427 WARN  [org.tes.doc.DockerClientProviderStrategy] (build-13) DOCKER_HOST unix:///Users/martin.grammelspacher/.docker/run/docker.sock is not listening
2023-07-17 21:57:03,431 INFO  [org.tes.doc.DockerMachineClientProviderStrategy] (build-13) docker-machine executable was not found on PATH (....)
2023-07-17 21:57:03,432 ERROR [org.tes.doc.DockerClientProviderStrategy] (build-13) Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
As no valid configuration was found, execution cannot continue.
See https://www.testcontainers.org/on_failure.html for more details.
2023-07-17 21:57:04,115 WARN  [io.qua.run.uti.ContainerRuntimeUtil] (build-13) Command "docker info" exited with error code 1. Rootless container runtime detection might not be reliable or the container service is not running at all.
2023-07-17 21:57:04,125 INFO  [org.tes.uti.ImageNameSubstitutor] (build-13) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2023-07-17 21:57:04,132 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2023-07-17 21:57:04,160 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.amazon.common.deployment.DevServicesLocalStackProcessor#startLocalStackDevService threw an exception: java.lang.RuntimeException: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
        at io.quarkus.amazon.common.deployment.DevServicesLocalStackProcessor.startLocalStack(DevServicesLocalStackProcessor.java:333)
        at io.quarkus.amazon.common.deployment.DevServicesLocalStackProcessor.lambda$startLocalStackDevService$8(DevServicesLocalStackProcessor.java:145)
        at java.base/java.util.HashMap.forEach(HashMap.java:1421)
        at io.quarkus.amazon.common.deployment.DevServicesLocalStackProcessor.startLocalStackDevService(DevServicesLocalStackProcessor.java:144)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:833)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
        at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:231)
        at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
        at org.testcontainers.DockerClientFactory.getTransportConfig(DockerClientFactory.java:156)
        at org.testcontainers.DockerClientFactory.getRemoteDockerUnixSocketPath(DockerClientFactory.java:171)
        at org.testcontainers.containers.localstack.LocalStackContainer.<init>(LocalStackContainer.java:112)
        at org.testcontainers.containers.localstack.LocalStackContainer.<init>(LocalStackContainer.java:94)
        at io.quarkus.amazon.common.deployment.DevServicesLocalStackProcessor.lambda$startLocalStack$25(DevServicesLocalStackProcessor.java:289)


```
Possible solutions:
- catch the exception, log a warning & startup anyways
- change the doc :sweat_smile:

Yeah, I know...setting `quarkus.devservices.enabled=false` is not a big issue. ^^

But with https://github.com/quarkusio/quarkus/issues/34801 we have to disable each service (which is a little bit noisy)
```
quarkus.s3.devservices.enabled=false
quarkus.sns.devservices.enabled=false
quarkus.sqs.devservices.enabled=false
quarkus.dynamodb.devservices.enabled=false
quarkus.ssm.devservices.enabled=false
```

--- a/docs/modules/ROOT/pages/dev-services.adoc
+++ b/docs/modules/ROOT/pages/dev-services.adoc
@@ -9,7 +9,7 @@ Dev Services for Amazon Services is automatically enabled for each extensions ad
 * `quarkus.devservices.enabled` is set to `false`
 * `devservices.enabled` is set to `false` per extension (eg. `quarkus.s3.devservices.enabled=false`) 
 * the `endpoint-override` is configured  (eg. `quarkus.s3.endpoint-override=http://localhost:4566`)
-* Your environment does not support Docker
+* Your environment does not support Docker. In test mode, it will cause the test to fail. Either disable Dev Services or configure the `endpoint-override`.
 
 == Shared services
 

